sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/implementation/models/BlobChangefeedEvent.java

```diff
@@ -0,0 +1,212 @@
+     * @param subject The subject.
+     * @param eventType {@link BlobChangefeedEventType}
+     * @param eventTime The {@link OffsetDateTime event time}.
+     * @param id The identifier.
+     * @param data {@link BlobChangefeedEventData}
+     * @param dataVersion The data version.
+     * @param metadataVersion The metadata version.
+     */
+    public BlobChangefeedEvent(String topic, String subject, BlobChangefeedEventType eventType,
+        OffsetDateTime eventTime, String id, com.azure.storage.blob.changefeed.models.BlobChangefeedEventData data,
+        Long dataVersion, String metadataVersion) {
+        this.topic = topic;
+        this.subject = subject;
+        this.eventType = eventType;
+        this.eventTime = eventTime;
+        this.id = id;
+        this.data = data;
+        this.dataVersion = dataVersion;
+        this.metadataVersion = metadataVersion;
+    }
+
+    /**
+     * Constructs a {@link BlobChangefeedEvent}.
+     *
+     * @param record The record.
+     * @return The {@link BlobChangefeedEvent} representing the record.
+     * @throws IllegalArgumentException if the record is not valid.
+     */
+    public static BlobChangefeedEvent fromRecord(Object record) {
+        AvroSchema.checkType("r", record, Map.class);
+        Map<?, ?> r = (Map<?, ?>) record;
+
+        if (!r.get(AvroConstants.RECORD).equals("BlobChangeEvent")) {
+            throw new IllegalArgumentException("Not a valid BlobChangefeedEvent.");
+        }
+
+        Object topic = r.get("topic");
+        Object subject = r.get("subject");
+        Object eventType = r.get("eventType");
+        Object eventTime = r.get("eventTime");
+        Object id = r.get("id");
+        Object data = r.get("data");
+        Object dataVersion = r.get("dataVersion");
+        Object metadataVersion = r.get("metadataVersion");
+
+        return new BlobChangefeedEvent(
+            nullOrString("topic", topic),
+            nullOrString("subject", subject),
+            isNull(eventType) ? null
+                : BlobChangefeedEventType.fromString(BlobChangefeedEvent.nullOrString("eventType", eventType)),
+            isNull(eventTime) ? null
+                : OffsetDateTime.parse(Objects.requireNonNull(nullOrString("eventTime", eventTime))),
+            nullOrString("id", id),
+            isNull(data) ? null : BlobChangefeedEventData.fromRecord(data),
+            nullOrLong("dataVersion", dataVersion),
+            nullOrString("metadataVersion", metadataVersion)
+        );
+    }
+
+    /**
+     * Determines whether or not the object is null in the Avro sense.
+     */
+    static boolean isNull(Object o) {
+        return o == null || o instanceof AvroNullSchema.Null;
+    }
+
+    /**
+     * Returns either null or a String.
+     */
+    static String nullOrString(String name, Object o) {
+        if (isNull(o)) {
+            return null;
+        }
+        AvroSchema.checkType(name, o, String.class);
+        return (String) o;
+    }
```

Comment:
all of this could be 
```
static <T> T nullOr(String name, Object o, Class<T> clazz)
```

See https://docs.oracle.com/javase/tutorial/extra/generics/methods.html